pkg/config/config.go

```diff
@@ -295,6 +304,7 @@ func NewConfigFromMap(data map[string]string) (*Config, error) {
 		cm.AsBool(EnableMeshPodAddressabilityKey, &nc.EnableMeshPodAddressability),
 		cm.AsString(DefaultExternalSchemeKey, &nc.DefaultExternalScheme),
 		cm.AsBool(InternalEncryptionKey, &nc.InternalEncryption),
+		cm.AsString(InternalTrustKey, &nc.InternalTrust),
```

Comment:
Not your fault, but I wish we had `AsEnum` here.

(Don't take action on this gripe in this PR)